CMakeLists.txt

```diff
@@ -145,6 +145,23 @@ if(CLIENTS OR TESTS)
   endif()
 endif()
 
+# Check plugin support
+find_path(DLFCN_H "dlfcn.h")
+if(DLFCN_H)
+  message(STATUS "Found <dlfcn.h>, enabling plugins using POSIX dynamic linker interface")
+  set(PLUGINS TRUE)
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_posix.cpp
+  )
+endif()
+
+if(NOT PLUGINS)
+  message(STATUS "Disabling plugin support")
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_none.cpp
+  )
+endif()
+
```

Comment:
Well, `Database` inherits from `plugin::Database`, and the whole idea is to use a stub plugin entry (that does not override anything) if no actual plugin is found instead of cluttering glue code with null pointer checks. So, unless we #ifdef out everything plugin-related, we cannot skip compilation of plugin-related sources (though we can avoid installing the extra header, of course).